nav2_mppi_controller/src/parameters_handler.cpp

```diff
@@ -64,17 +68,23 @@ ParametersHandler::dynamicParamsCallback(
     if (auto callback = get_param_callbacks_.find(param_name);
       callback != get_param_callbacks_.end())
     {
-      callback->second(param);
+      callback->second(param, result);
     } else {
-      RCLCPP_WARN(logger_, "Parameter %s not found", param_name.c_str());
+      result.successful = false;
```

Comment:
Wouldn't multiple `parameters` in the for loop overwrite the `result.successful = false` if the next one is successful?

Reply:
That is not my read of it.  

The result.successful is initialized to true outside the for loop.

The "contract" (albeit un-enforceable) of the callback is to only touch the  reference to the result passed into it on failure.

Reply:
This is true, sorry

Reply:
No worries.

Question:  Is the "atomically" in "set_parameters_atomically" intended to mean?

1. Change multiple parameters in the same operation (tick)  OR 
2. Only change parameters if all parameters can be set without failure

The current main branch AND this PR (currently) supports option 1.

Better support for option 2  could be offered by passing a reference to the result to the pre_callbacks where validation could occur.

For instance, Rejection of attempts to change a static parameter type should probably be done in a pre_callback.

Then a check for result.successful should be done before looping again and calling the callbacks to actually set the parameters.  These callbacks could also fail, but at that point supporting a rollback is going to add a lot more complexity.

If option 1 is enough then I think is ready for review.

Reply:
Atomically means its all or nothing for multiple parameters, while non-atomically sets up to the failure.

I think option 1 is fine for now. We don't really handle dynamic parameters in the way that the atomic would make a difference. Even if the `rclcpp::get_parameter()` returns the original value due to a failure, the stored `my_param_value_` would be updated in the dynamic parameter callbacks for each case. 

I think to properly handle the dynamic parameter callbacks, we'd need the dynamic parameter callbacks to **only** do parameter validation, then as a post-callback step, re-get all parameter values via `get_parameter`. That would require a bit update across the stack, not just in MPPI, to be treated that way.

We could make that update, but I don't think we need to make that here in this PR. I'm a fan of multiple bitesized PRs versus one very large one.